actionpack/lib/action_dispatch/http/url.rb

```diff
@@ -88,7 +88,11 @@ def add_params(path, params)
 
         def add_anchor(path, anchor)
           if anchor
-            path << "##{Journey::Router::Utils.escape_fragment(anchor.to_param)}"
+            param = anchor.to_param
+            unless Journey::Router::Utils.escaped_fragment?(param)
```

Comment:
Hmmm... this seems reasonable but could it introduce any potential security vulnerabilities if we're conditionally escaping?

Reply:
Is this operation idempotent? If so, we could forsake performance for security and just always call the action.